pytroll_collectors/tests/test_geographic_gatherer.py

```diff
@@ -188,8 +211,13 @@ def test_init_posttroll(self):
                 dt.timedelta(seconds=12, microseconds=300000), None, None) in self.RegionCollector.mock_calls
 
         # A publisher is created with composed name and started
-        self.publisher.NoisyPublisher.assert_called_once_with('gatherer_'+'_'.join(sections), port=0, nameservers=None)
-        self.publisher.NoisyPublisher.return_value.start.assert_called_once()
+        expected = {
+            'name': 'gatherer_'+'_'.join(sections),
+            'port': 0,
+            'nameservers': None,
+        }
+        self.create_publisher_from_dict_config.assert_called_once_with(expected)
+        self.create_publisher_from_dict_config.return_value.start.assert_called_once()
```

Comment:
this code is duplicated 5 times, can it be refactored into a function/method that takes `name`, `port` and `nameservers` as arguments?

Reply:
Done